src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/config/HmppsNonAssociationsApiExceptionHandler.kt

```diff
@@ -104,6 +118,9 @@ class HmppsNonAssociationsApiExceptionHandler {
   }
 }
 
+class NonAssociationNotFoundException(id: Long) :
```

Comment:
I gave it a try but it complained about having a private constructor or something when I tried to instantiate that same exception class 🤷

Reply:
tbh I'm a bit confused but how many exceptions classes are there 😅 

I may try and just handle `ResponseStatusException`s [as it's done here on SO](https://stackoverflow.com/a/54539641).

Let me push a commit and if you don't like we can always revert but it may be more generic.

Reply:
Does this make more sense? https://github.com/ministryofjustice/hmpps-non-associations-api/pull/33/commits/a026bebc60412dd054bccda9eed60ffa968a470f